test/unit/recipes/create_pwfile_spec.rb

```diff
@@ -0,0 +1,34 @@
+# encoding: UTF-8
+#
+# Author:: Xabier de Zuazo (<xabier@zuazo.org>)
+# Copyright:: Copyright (c) 2014 Onddo Labs, SL.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative '../spec_helper'
+
+describe 'dovecot::create_pwfile', order: :random do
+  let(:chef_runner) { ChefSpec::SoloRunner.new }
+  let(:chef_run) { chef_runner.converge(described_recipe) }
+  let(:node) { chef_runner.node }
+
+  context 'when the module runs' do
+    it 'creates a file' do
+      expect(chef_run). to create_template('/etc/dovecot/password')
+        .with_path('/etc/dovecot/password')
+        .with_source('password.erb')
+    end
+  end
```

Comment:
unit tests... the worst for me :( I cannot get it yet (though pretty new with these concepts)